lib/FrameManager.js

```diff
@@ -672,7 +672,7 @@ class Frame {
   waitForFunction(pageFunction, options = {}, ...args) {
     const timeout = helper.isNumber(options.timeout) ? options.timeout : 30000;
     const polling = options.polling || 'raf';
-    return new WaitTask(this, pageFunction, polling, timeout, ...args).promise;
+    return new WaitTask(this, pageFunction, options.title || 'function', polling, timeout, ...args).promise;
```

Comment:
let's not add `options.title`; this will be an outlier in our api. Just passing in `"function"`, or maybe stringified function body is enough.

Reply:
I see. I'd prefer copying the three lines of `waitForFunction` in `__waitForSelectorOrXPath` method.